### PR TITLE
[full-ci] Update idp UI

### PR DESCRIPTION
**Pull request by C0rby**
## Description

Updated our fork of the lico IdP UI. This also updated the used npm dependencies.
The design didn't change.

I also added the [ODC-By-1.0 license](https://opendatacommons.org/licenses/by/1-0/) to the license check

## Related Issue

- Fixes https://github.com/owncloud/ocis/issues/3493

## Motivation and Context

We need to reduce the amount of vulnerable js libraries.

## How Has This Been Tested?

- locally

## Screenshots (if appropriate):
Looks like before.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Code changes

### CODE REVIEW

**Comment by C0rby** [comment]
Removing the 'react-scripts' dependency would resolve the license issue but then we need to add a bunch of webpack, babel, plugins etc. I think the easiest approach would be to take the versions from the react-scripts v4.0.3 package.json.

File: services/idp/.eslintrc.json.old [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (pascalwengerter) [inline services/idp/.eslintrc.json.old:1]: Can this file be removed/merged with what is in the package.json?

File: services/idp/package.json.old [hunk 2]
```diff
@@ -0,0 +1,156 @@
+{
```
Comment (pascalwengerter) [inline services/idp/package.json.old:1]: Old package.json can be removed?

File: services/idp/package.json [hunk 3]
```diff
@@ -6,151 +6,145 @@
...
+    "@testing-library/jest-dom": "^5.11.4",
+    "@testing-library/react": "^11.1.0",
+    "@testing-library/user-event": "^12.1.10",
+    "@types/jest": "^26.0.15",
+    "@types/node": "^12.0.0",
+    "@types/react": "^17.0.2",
+    "@types/react-dom": "^17.0.2",
+    "@types/react-redux": "^7.1.19",
+    "@types/redux-logger": "^3.0.9",
+    "axios": "^0.22.0",
```
Comment (pascalwengerter) [inline services/idp/package.json:79]: Could be bumped to v0.27.x?

File: services/idp/package.json [hunk 4]
```diff
@@ -6,151 +6,145 @@
...
+    "react-redux": "^7.2.5",
+    "react-router": "^5.2.1",
+    "react-router-dom": "5.2.1",
+    "redux": "^3.7.2",
     "redux-logger": "^3.0.6",
     "redux-thunk": "^2.2.0",
     "render-if": "^0.1.1",
-    "semver": "7.3.4",
-    "typeface-roboto": "^1.1.13"
+    "typescript": "^4.1.2",
```
Comment (pascalwengerter) [inline services/idp/package.json:98]: Dev dependency :)

File: services/idp/package.json [hunk 5]
```diff
@@ -6,151 +6,145 @@
...
-    "webpack-dev-server": "^3.11.2",
-    "webpack-manifest-plugin": "3.1.0",
-    "workbox-webpack-plugin": "4.3.1",
-    "yarn-deduplicate": "^3.1.0"
-  },
-  "packageManager": "yarn@3.1.0"
+    "webpack": "4.44.2",
+    "webpack-manifest-plugin": "4.1.1",
+    "workbox-webpack-plugin": "5.1.4",
+    "yarn-deduplicate": "^1.1.1"
```
Comment (pascalwengerter) [inline services/idp/package.json:148]: Can be safely removed IIRC

File: services/idp/src/reducers/common.js [hunk 6]
```diff
@@ -20,24 +20,14 @@ const defaultPathPrefix = (() => {
...
-
 const defaultState = {
   hello: null,
+  branding: null,
   error: null,
   flow: flow,
   query: query,
   updateAvailable: false,
-  pathPrefix: defaultPathPrefix,
-  passwordResetLink: defaultPasswordResetLink
```
Comment (pascalwengerter) [inline services/idp/src/reducers/common.js:40]: Are we loosing the functionality you did build in with the `passwordResetLink` here?

**Review by pascalwengerter — COMMENTED** [review]
Only small remarks, pretty neat already!

**Comment by wkloucek** [comment]
@dragotin can we use the ODC-By-1.0 license? 
https://joinup.ec.europa.eu/licence/compatibility-check/ODC-By-1.0/Apache-2.0

**Review by wkloucek — APPROVED** [review]
Looks good from a users' perspective. Though I don't know if we need to care about something when ODC-By-1.0.

**Comment by C0rby** [comment]
> @dragotin can we use the ODC-By-1.0 license? https://joinup.ec.europa.eu/licence/compatibility-check/ODC-By-1.0/Apache-2.0

AFAIU, we can safely use stuff licensed under ODC-By-1.0

**Comment by wkloucek** [comment]
> > @dragotin can we use the ODC-By-1.0 license? https://joinup.ec.europa.eu/licence/compatibility-check/ODC-By-1.0/Apache-2.0
> 
> AFAIU, we can safely use stuff licensed under ODC-By-1.0

OK, then let's merge

**Pull request merged**